README.md

```diff
@@ -1,50 +1,56 @@
-- run   `go run main.go`
-- congrats! you've just made your first sysl application!
+- run `make gen`. 
+  - This generates the Go code.
+- run `make build`. 
+  - This builds the generated code using the installed Go runtime.
+- run `make run`.
+  - This runs the petdemo app and the downstream apps i.e. flickr and petstore
+- `curl http://localhost:8080/random-pet-pic` or open in browser.
+  - Output: `{"name":"white golden retriever","uri":"http://www.example.com/dog/wgr"}`
 
+Congrats! You've just built and run the sysl-go demo application!
 
 ## Development
-- Refer to the [Makefile](Makefile) to generate client and server code for all of your applications
 
-## File structure 
+- Refer to the [Makefile](Makefile) to generate client and server code for all of your applications.
 
-api/: contains all API specifications for the generated application
+## File structure
 
-gen/: contains all the generated code for the service
+[internal/gen/petdemo]: Contains the all the generated code for the petdemo app.
+[internal/gen/flickr]: Contains the generated code for the flickr downstream service.
+[internal/gen/petstore]: Contains the generated code for the petstore downstream service.
 
-[internal/server/server.go](pkg/server/server.go): The hand-written code that's written; Server config and such
+[internal/petdemo]: Hand crafted config and server for petdemo app.
+[internal/petdemo/handlers]: Hand crafted handlers for the petdemo app.
 
-pkg/defaultcallback: contains code that sets up the defaults for generated code. (This will no longer be necessary in future Sysl versions)
+[internal/flickr]: Hand crafted config and server for flickr downstream service.
+[internal/flickr/handlers]: Hand crafted handlers for the flickr downstream service.
 
-When new endpoints are added, they need to be added to the `simple.ServiceInterface` variable in [server.go](server/server.go)
+[internal/petstore]: Hand crafted config and server for petstore downstream service.
+[internal/petstore/handlers]: Hand crafted handlers for the petstore downstream service.
 
-[main.go](main.go): runs the actual server
+## Configuration
 
+The configuration is driven through the YAML files in the repository root.
+- [config-flickr.yml](config-flickr.yml)
+- [config-petdemo.yaml](config-petdemo.yaml)
+- [config-petstore.yaml](config-petstore.yaml)
+The arrai and sysl versions can be locked in the arraiw.properties and syslw.properties files in repository root.
 
-run `make` to regenerate application code
-First you need to edit the start of the Makefile:
+## Application specs
 
-```
-input = your input sysl file
-app = < the app you want to develop>
-down = <downstreams in a list separated by spaces>
-basepath = <Your current go module path>
-```
-
-so: `make input=model/simple.sysl app=Simple` for this example
-
-run `go run main.go` to start the server
+The application specs are stored in the form of yaml files in specs directory. 
+- specs/backend/flickr/flickr.yaml
```

Comment:
Unresolved; link these as well?

Reply:
Didn't intend to make it a link. The file is present is the same repo.